Start of Overview Page Rework (#5728)
nautobot/nautobot: nautobot/docs/overview/index.md

# 
<figure markdown="span">
  ![Nautobot Logo](../assets/nautobot_logo.svg "Welcome"){ width="400" }

>   <figcaption></figcaption>

Reviewer: Is this needed even as a blank tag, or can it be removed?

2. **Extensible Data Platform for Automation** - Nautobot has a rich feature set to seamlessly integrate with network automation solutions.  Nautobot offers GraphQL and native Git integration along with REST APIs and webhooks.  Git integration dynamically loads YAML data files as Nautobot config contexts.  Nautobot also has an evolving App system that enables users to create custom models, APIs, and UI elements.  The App system is also used to unify and aggregate disparate data sources creating a Single Source of Truth to streamline data management for network automation.

3. **Platform for Network Automation Apps** - The Nautobot App system enables users to create Network Automation Apps.  Apps can be as lightweight or robust as needed based on user needs.  Using Nautobot for creating custom applications saves up to 70% development time by re-using features such as authentication, permissions, webhooks, GraphQL, change logging, etc. all while having access to the data already stored in Nautobot.

## Getting Started

<div class="grid cards" markdown>

- :fontawesome-solid-download:{ .lg .middle } **Installing Nautobot**
    
    ---

>     Get up and running with a [Nautobot install](../user-guide/administration/installation/index.md) on your own Linux VM or in a Docker environment

Reviewer: ```suggestion
    Get up and running with a [Nautobot install](../user-guide/administration/installation/index.md) on your own Linux VM or in a Docker environment.
```

- :fontawesome-solid-user-gear:{ .lg .middle } **Configuring Nautobot**
    
    ---

>     [Configuration of Nautobot](../user-guide/administration/configuration/index.md)

Reviewer: ```suggestion
    Learn about the many [configuration options](../user-guide/administration/configuration/index.md) that Nautobot offers for fine-tuning your installation.
```

> - :material-web-check:{ .lg .middle } **Getting Started with Nautobot** 

Reviewer: Not a big fan of mixing-and-matching material icons and fontawesome icons. Can we use all material icons?

    
    ---
    Dive into [how to use Nautobot](../user-guide/feature-guides/getting-started/index.md) and the key components of the core web interface.

>     How [Nautobot Apps](../apps/index.md) can expand Nautobot's functionality.

Reviewer: Is this line redundant with the "Nautobot Apps" card immediately below?

- :material-apps:{ .lg .middle } **Nautobot Apps**

    ---
    Nautobot has a thriving ecosystem of **Apps**, developed as separate projects, for which you can find links to documentation under the [Nautobot Apps](../apps/index.md) section.

-  :material-api:{ .lg .middle } **Nautobot APIs!**

    ---

>     Dive into the [REST](../user-guide/platform-functionality/rest-api/overview.md), [GraphQL](../user-guide/platform-functionality/graphql.md) APIs

Reviewer: ```suggestion
    Dive into the [REST](../user-guide/platform-functionality/rest-api/overview.md) and [GraphQL](../user-guide/platform-functionality/graphql.md) APIs.
```

-  :material-language-python:{ .lg .middle } **Nautobot SDKs**

    ---
    Nautobot has a [Python SDK](https://pynautobot.readthedocs.io/) and [Ansible modules](https://galaxy.ansible.com/ui/repo/published/networktocode/nautobot/docs/) to interact with Nautobot in a programmatic way.

</div>

## Nautobot Apps

<div class="grid cards" markdown>

- ![Golden Config](../assets/app-icons/icon-GoldenConfiguration.png){style="height: 35px; margin-bottom: 0em" .middle } **Golden Configuration**

    ---

>     [Golden Configuration](https://github.com/nautobot/nautobot-app-golden-config) backs up configurations, generate intended state configurations, compare for compliance and remediate device configurations.

Reviewer: ```suggestion
    [Golden Configuration](https://github.com/nautobot/nautobot-app-golden-config) backs up configurations, generates intended state configurations, compares them for compliance and remediates device configurations.
```

- ![Device Lifecycle](../assets/app-icons/icon-DeviceLifecycle.png){style="height: 35px; margin-bottom: 0em" .middle } **Device Lifecycle**

    ---

>     [Device Lifecycle](https://github.com/nautobot/nautobot-app-device-lifecycle-mgmt) adds additional capabilities around managing **hardware** lifecycle, **software** images, and **contracts** around your systems.

Reviewer: ```suggestion
    [Device Lifecycle](https://github.com/nautobot/nautobot-app-device-lifecycle-mgmt) adds additional capabilities around managing the **hardware** and **software** lifecycle, including the tracking of related **contracts** .
```

- ![Firewall Models](../assets/app-icons/icon-FirewallModels.png){style="height: 35px; margin-bottom: 0em" .middle } **Firewall Models**

    ---

>     [Firewall Models](https://github.com/nautobot/nautobot-app-firewall-models) works to help model out firewall rules and objects. Using Nautobot to help generically manage firewall rules, including extended ACLs.

Reviewer: ```suggestion
    [Firewall Models](https://github.com/nautobot/nautobot-app-firewall-models) helps to model out firewall rules and related objects, including extended ACLs.
```

- ![SSoT](../assets/app-icons/icon-SSoT.png){style="height: 35px; margin-bottom: 0em" .middle } **SSoT**

    ---

>     [SSoT](https://github.com/nautobot/nautobot-app-ssot) is the framework to synchronize data from additional sources into and out of Nautobot, powered by DiffSync.

Reviewer: ```suggestion
    [Single Source of Truth](https://github.com/nautobot/nautobot-app-ssot) is the framework to synchronize data from other systems into and out of Nautobot.
```

 - ![ChatOps](../assets/app-icons/icon-ChatOps.png){style="height: 35px; margin-bottom: 0em" .middle } **ChatOps**

    ---

>     [ChatOps](https://github.com/nautobot/nautobot-app-chatops) allows you to quickly get up and creating ChatOps commands. Start to allow peer teams to interact and get information about the network.

Reviewer: ```suggestion
    [ChatOps](https://github.com/nautobot/nautobot-app-chatops) supports a variety of chat applications, allowing peer teams to conveniently interact with Nautobot and get information about the network.
```

 - ![Circuit Maintenance](../assets/app-icons/icon-nautobot-circuit-maintenance.png){style="height: 35px; margin-bottom: 0em" .middle } **Circuit Maintenance**

    ---
    [Circuit Maintenance](https://github.com/nautobot/nautobot-app-circuit-maintenance) brings your circuit maintenance notification emails (and API connected info) into objects within Nautobot to bring better notification and business actions to the maintenances.

 - ![Capacity Metrics](../assets/app-icons/icon-nautobot-capacity-metrics.svg){style="height: 35px; margin-bottom: 0em" .middle } **Capacity Metrics**

    ---

>     [Capacity Metrics](https://github.com/nautobot/nautobot-app-capacity-metrics) brings the Nautobot data to the Prometheus metrics endpoints, giving you time series information about your Nautobot data.

Reviewer: ```suggestion
    [Capacity Metrics](https://github.com/nautobot/nautobot-app-capacity-metrics) brings additional Nautobot data to Prometheus metrics, making it easy to derive time series information about your Nautobot data.
```

 - ![Device Onboarding](../assets/app-icons/icon-DeviceOnboarding.png){style="height: 35px; margin-bottom: 0em" .middle } **Device Onboarding**

    ---

>     [Device Onboarding](https://github.com/nautobot/nautobot-app-device-onboarding) brings the network data into your SOT, helping to build out the intended state from the current state of the network.

Reviewer: ```suggestion
    [Device Onboarding](https://github.com/nautobot/nautobot-app-device-onboarding) brings network data into Nautobot, helping to build out the intended state from the current state of the network.
```

</div>

## Nautobot Screenshots

<div class="grid cards" markdown>

- ![Main Page](../media/ss_main_page.png)

- ![Config Contexts](../media/ss_config_contexts.png)

- ![Prefix Hierarchy](../media/ss_prefix_hierarchy.png)

- ![GraphQL API](../media/ss_graphiql.png)

</div>

### App Screenshots

<div class="grid cards" markdown>

- ![Golden Config App](../media/ss_app_golden_config.png)

- ![ChatOps App](../media/ss_app_chatops.png)

</div>

## Contributing to Nautobot

<div class="grid cards" markdown>

- :fontawesome-solid-code:{ .lg .middle } **Develop and Contribute to Nautobot**
    
    ---
    Learn how to [develop and contribute to Nautobot](../development/index.md)

- :fontawesome-solid-user-gear:{ .lg .middle } **Set Up Your Developer Environment**
    
    ---

>     [Configuration of Nautobot](../user-guide/administration/configuration/index.md)

Reviewer: Wrong content?

- :material-web-check:{ .lg .middle } **Development Guide** 
    
    ---
    Dive into [how to use Nautobot](../user-guide/feature-guides/getting-started/index.md) and the key components of the core web interface.

>     How [Nautobot Apps](../apps/index.md) can expand Nautobot's functionality.

Reviewer: Wrong content?

1. **Flexible Source of Truth for Networking** - Nautobot core data models are used to define the intended state of network infrastructure enabling it as a Source of Truth. While a baseline set of models are provided (such as IP networks and addresses, devices and racks, circuits and cable, etc.) it is Nautobot's goal to offer maximum data model flexibility. This is enabled through features such as user-defined relationships, custom fields on any model, and data validation that permits users to codify everything from naming standards to having automated tests run before data can be populated into Nautobot.

2. **Extensible Data Platform for Automation** - Nautobot has a rich feature set to seamlessly integrate with network automation solutions.  Nautobot offers GraphQL and native Git integration along with REST APIs and webhooks.  Git integration dynamically loads YAML data files as Nautobot config contexts.  Nautobot also has an evolving App system that enables users to create custom models, APIs, and UI elements.  The App system is also used to unify and aggregate disparate data sources creating a Single Source of Truth to streamline data management for network automation.

> 3. **Platform for Network Automation Apps** - The Nautobot App system enables users to create Network Automation Apps.  Apps can be as lightweight or robust as needed based on user needs.  Using Nautobot for creating custom applications saves up to 70% development time by re-using features such as authentication, permissions, webhooks, GraphQL, change logging, etc. all while having access to the data already stored in Nautobot.

Author: I'll see if ChatGPT can help.

> 3. **Platform for Network Automation Apps** - The Nautobot App system enables users to create Network Automation Apps.  Apps can be as lightweight or robust as needed based on user needs.  Using Nautobot for creating custom applications saves up to 70% development time by re-using features such as authentication, permissions, webhooks, GraphQL, change logging, etc. all while having access to the data already stored in Nautobot.

Author: Give the new one a read @glennmatthews.

## Getting Started

<!-- markdownlint-disable MD033 -->
<div class="grid cards" markdown>

- :material-cloud-download:{ .lg .middle } **Installing Nautobot**

    ---
    Get up and running with a [Nautobot install](../user-guide/administration/installation/index.md) on your own Linux VM or in a Docker environment.

- :material-account-settings:{ .lg .middle } **Configuring Nautobot**

    ---
    Learn about the many [configuration options](../user-guide/administration/configuration/index.md) that Nautobot offers for fine-tuning your installation.

- :material-web-check:{ .lg .middle } **Getting Started with Nautobot**

    ---
    Dive into [how to use Nautobot](../user-guide/feature-guides/getting-started/index.md) and the key components of the core web interface.
    How [Nautobot Apps](../apps/index.md) can expand Nautobot's functionality.

    ---

- :material-api:{ .lg .middle } **Nautobot APIs!**

    ---
    Dive into the [REST](../user-guide/platform-functionality/rest-api/overview.md) and [GraphQL](../user-guide/platform-functionality/graphql.md) APIs.

- :material-language-python:{ .lg .middle } **Nautobot SDKs**

    ---

>     Nautobot has a [Python SDK](https://pynautobot.readthedocs.io/) and [Ansible modules](https://galaxy.ansible.com/ui/repo/published/networktocode/nautobot/docs/) to interact with Nautobot in a programmatic way.

Author: Hmm, will check. I made the change to use the mkdocs set up. Not sure about the rest.

>     Nautobot has a [Python SDK](https://pynautobot.readthedocs.io/) and [Ansible modules](https://galaxy.ansible.com/ui/repo/published/networktocode/nautobot/docs/) to interact with Nautobot in a programmatic way.

Reviewer: https://docs.nautobot.com/projects/pynautobot/en/latest/index.html

</div>
<!-- markdownlint-enable MD033 -->

## Nautobot Apps

Nautobot has a thriving ecosystem of **Apps**, developed as separate projects, for which you can find links to documentation under the [Nautobot Apps](../apps/index.md) section.

<!-- markdownlint-disable MD033 -->

> <div class="grid cards" markdown>

Author: I can apply there as well, but I think this makes sense to keep it here. The extensibility messaging is a big part of why Nautobot.

#
<!-- markdownlint-disable no-inline-html -->
<figure markdown="span">
  ![Nautobot Logo](../assets/nautobot_logo.svg "Welcome"){ width="400" }
</figure>
<!-- markdownlint-enable no-inline-html -->
---

## What is Nautobot?

Nautobot is an open source Network Source of Truth and Network Automation Platform built as a web application atop the [Django](https://www.djangoproject.com/) Python framework with a [PostgreSQL](https://www.postgresql.org/) or [MySQL](https://www.mysql.com) database.

Try out a live demo at [https://demo.nautobot.com](https://demo.nautobot.com)

## Key Use Cases

Nautobot enables three (3) key use cases.

> 1. **Flexible Source of Truth for Networking** Nautobot serves as a highly flexible source of truth for network infrastructure, utilizing its core data models to define the intended state of your network. While it provides a baseline set of models such as IP networks, devices, racks, circuits, and cables, Nautobot's true strength lies in its flexibility. Users can define custom relationships and fields for any model, ensuring that the platform adapts to their specific needs. Additionally, Nautobot's robust data validation allows for the enforcement of naming standards and the execution of automated tests before data is committed, ensuring data integrity and consistency.
> 
> 2. **Extensible Data Platform for Automation** Nautobot is designed to seamlessly integrate with network automation solutions, offering a comprehensive feature set that includes GraphQL, native Git integration, REST APIs, and webhooks. The Git integration allows Nautobot to dynamically load YAML data files as configuration contexts. The evolving App system further enhances Nautobot's capabilities, enabling users to create custom models, APIs, and UI elements. By unifying and aggregating disparate data sources, Nautobot provides a single source of truth, simplifying data management and streamlining network automation processes.
> 
> 3. **Platform for Network Automation Apps** The Nautobot App system empowers users to develop custom Network Automation Apps tailored to their specific needs. Whether lightweight or complex, these apps benefit from Nautobot's existing features such as authentication, permissions, webhooks, GraphQL, and change logging, significantly reducing development time by up to 70%. This integration not only accelerates the development process but also leverages the extensive data already stored in Nautobot, providing a powerful and efficient platform for building network automation solutions.

Reviewer: Wording feels better to me, but I'd prefer to keep the dashes between the titles and details for readability:

```suggestion
1. **Flexible Source of Truth for Networking** — Nautobot serves as a highly flexible source of truth for network infrastructure, utilizing its core data models to define the intended state of your network. While it provides a baseline set of models such as IP networks, devices, racks, circuits, and cables, Nautobot's true strength lies in its flexibility. Users can define custom relationships and fields for any model, ensuring that the platform adapts to their specific needs. Additionally, Nautobot's robust data validation allows for the enforcement of naming standards and the execution of automated tests before data is committed, ensuring data integrity and consistency.

2. **Extensible Data Platform for Automation** — Nautobot is designed to seamlessly integrate with network automation solutions, offering a comprehensive feature set that includes GraphQL, native Git integration, REST APIs, and webhooks. The Git integration allows Nautobot to dynamically load YAML data files as configuration contexts. The evolving App system further enhances Nautobot's capabilities, enabling users to create custom models, APIs, and UI elements. By unifying and aggregating disparate data sources, Nautobot provides a single source of truth, simplifying data management and streamlining network automation processes.

3. **Platform for Network Automation Apps** — The Nautobot App system empowers users to develop custom Network Automation Apps tailored to their specific needs. Whether lightweight or complex, these apps benefit from Nautobot's existing features such as authentication, permissions, webhooks, GraphQL, and change logging, significantly reducing development time by up to 70%. This integration not only accelerates the development process but also leverages the extensive data already stored in Nautobot, providing a powerful and efficient platform for building network automation solutions.
```

## Getting Started

<!-- markdownlint-disable no-inline-html -->
<div class="grid cards" markdown>

- :material-cloud-download:{ .lg .middle } **Installing Nautobot**

    ---
    Get up and running with a [Nautobot install](../user-guide/administration/installation/index.md) on your own Linux VM or in a Docker environment.

- :material-cog:{ .lg .middle } **Configuring Nautobot**

    ---
    Learn about the many [configuration options](../user-guide/administration/configuration/index.md) that Nautobot offers for fine-tuning your installation.

> - :material-play-network:{ .lg .middle } **Getting Started with Nautobot**

Reviewer: Maybe a better parallel to "Installing" and "Configuring":

```suggestion
- :material-play-network:{ .lg .middle } **Using Nautobot**
```

    Learn how [Nautobot Apps](../apps/index.md) can expand Nautobot's functionality.

    ---

- :material-api:{ .lg .middle } **Nautobot APIs!**

    ---
    Dive into the [REST](../user-guide/platform-functionality/rest-api/overview.md) and [GraphQL](../user-guide/platform-functionality/graphql.md) APIs.

- :material-language-python:{ .lg .middle } **Nautobot SDKs**

    ---
    Nautobot has a [Python SDK](https://pynautobot.readthedocs.io/) and [Ansible modules](https://galaxy.ansible.com/ui/repo/published/networktocode/nautobot/docs/) to interact with Nautobot in a programmatic way.

</div>
<!-- markdownlint-enable no-inline-html -->

## Nautobot Apps

Nautobot has a thriving ecosystem of **Apps**, developed as separate projects, for which you can find links to documentation under the [Nautobot Apps](../apps/index.md) section.

<!-- markdownlint-disable no-inline-html -->
<div class="grid cards" markdown>

- ![Golden Config](../assets/app-icons/icon-GoldenConfiguration.png){style="height: 35px; margin-bottom: 0em" .middle } **Golden Configuration**

    ---

>     [Golden Configuration](https://github.com/nautobot/nautobot-app-golden-config) backs up configurations, generates intended state configurations, compares them for compliance and remediates device configurations.

Reviewer: Feels like these links should point to the docs for each app instead of to their github repos?

    ---
    [Device Lifecycle](https://github.com/nautobot/nautobot-app-device-lifecycle-mgmt) adds additional capabilities around managing the **hardware** and **software** lifecycle, including the tracking of related **contracts** .

- ![Firewall Models](../assets/app-icons/icon-FirewallModels.png){style="height: 35px; margin-bottom: 0em" .middle } **Firewall Models**

    ---
    [Firewall Models](https://github.com/nautobot/nautobot-app-firewall-models) helps to model out firewall rules and related objects, including extended ACLs.

- ![SSoT](../assets/app-icons/icon-SSoT.png){style="height: 35px; margin-bottom: 0em" .middle } **SSoT**

    ---
    [Single Source of Truth](https://github.com/nautobot/nautobot-app-ssot) is the framework to synchronize data from other systems into and out of Nautobot.

- ![ChatOps](../assets/app-icons/icon-ChatOps.png){style="height: 35px; margin-bottom: 0em" .middle } **ChatOps**

    ---
    [ChatOps](https://github.com/nautobot/nautobot-app-chatops) supports a variety of chat applications, allowing peer teams to conveniently interact with Nautobot and get information about the network.

> - ![Circuit Maintenance](../assets/app-icons/icon-nautobot-circuit-maintenance.png){style="height: 35px; margin-bottom: 0em" .middle } **Circuit Maintenance**

Reviewer: Looks like for self-consistency this should be named `icon-CircuitMaintenance.png`

    ---
    [Circuit Maintenance](https://github.com/nautobot/nautobot-app-circuit-maintenance) brings your circuit maintenance notification emails (and API connected info) into objects within Nautobot to bring better notification and business actions to the maintenances.

> - ![Capacity Metrics](../assets/app-icons/icon-nautobot-capacity-metrics.svg){style="height: 35px; margin-bottom: 0em" .middle } **Capacity Metrics**

Reviewer: Looks like for self-consistency this should be named `icon-CapacityMetrics.svg`

- ![Device Onboarding](../assets/app-icons/icon-DeviceOnboarding.png){style="height: 35px; margin-bottom: 0em" .middle } **Device Onboarding**

    ---
    [Device Onboarding](https://github.com/nautobot/nautobot-app-device-onboarding) brings network data into Nautobot, helping to build out the intended state from the current state of the network.

</div>
<!-- markdownlint-enable no-inline-html -->

## Nautobot Screenshots

<!-- markdownlint-disable no-inline-html -->
<div class="grid cards" markdown>

- ![Main Page](../media/ss_main_page.png){ .on-glb }

- ![Config Contexts](../media/ss_config_contexts.png){ .on-glb }

- ![Prefix Hierarchy](../media/ss_prefix_hierarchy.png){ .on-glb }

- ![GraphQL API](../media/ss_graphiql.png){ .on-glb }

</div>
<!-- markdownlint-enable no-inline-html -->

### App Screenshots

<!-- markdownlint-disable no-inline-html -->
<div class="grid cards" markdown>

- ![Golden Config App](../media/ss_app_golden_config.png){ .on-glb }

- ![ChatOps App](../media/ss_app_chatops.png){ .on-glb }

</div>
<!-- markdownlint-enable no-inline-html -->

## Contributing to Nautobot

<!-- markdownlint-disable no-inline-html -->
<div class="grid cards" markdown>

> - :material-code-block-braces:{ .lg .middle } **Develop and Contribute to Nautobot**
> 
>     ---
>     Learn how to [develop and contribute to Nautobot](../development/index.md)
> 
> - :material-laptop:{ .lg .middle } **Set Up Your Developer Environment**
> 
>     ---
>     [Set up your Development Environment](../development/core/getting-started.md)
> 
> - :material-pier-crane:{ .lg .middle } **Jobs Development Guide**
> 
>     ---
>     Jump start your [development of Nautobot Jobs](../development/jobs/index.md).

Reviewer: Perhaps?

```suggestion
- :material-pier-crane:{ .lg .middle } **Jobs Developer Guide**

    ---
    Jump start your [development of Nautobot Jobs](../development/jobs/index.md).

- :material-application-brackets:{ .lg .middle } **Apps Developer Guide**

    ---
    Get started [developing Nautobot Apps](../development/apps/index.md)

- :material-file-code:{ .lg .middle } **Core Developer Guide**

    ---
    Learn how to [develop and contribute to Nautobot](../development/core/getting-started.md)
```
